### PR TITLE
Refactor server option

### PR DESCRIPTION
**Pull request by ralfbawg**
Refactor server option,pointer safe and enhance scalability

### CODE REVIEW

**Comment by googollee** [comment]
I hope it could keep compatible. We could update the API in new version.

File: server.go [hunk 1]
```diff
@@ -42,10 +43,10 @@ type Server struct {
 }
 
 // NewServer returns a server.
-func NewServer(c *engineio.Options) *Server {
+func NewServer(c ...engineio.Option) *Server {
```
Comment (mrfoe7) [inline server.go:46]: please, update `c` to `opts`

Comment (mrfoe7) [reply server.go:46]: it is important thing, people will use your API ))

File: _examples/default-http/main.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func main() {
-	server := socketio.NewServer(nil)
+	server := socketio.NewServer()
```
Comment (mrfoe7) [inline _examples/default-http/main.go:11]: we need add socketio option's wrapper that throw engine type and so on

**Comment by mrfoe7** [comment]
@ralfbawg please change target branch in PR to `v2-dev`

**Pull request merged**